pkg/cmd/server/api/validation/validation.go

```diff
@@ -86,6 +70,141 @@ func ValidateKubernetesMasterConfig(config *api.KubernetesMasterConfig) errs.Val
 	return allErrs
 }
 
+func ValidateOAuthConfig(config *api.OAuthConfig) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+
+	if len(config.ProxyCA) > 0 {
+		allErrs = append(allErrs, ValidateFile(config.ProxyCA, "proxyCA")...)
+	}
+
+	if len(config.MasterURL) == 0 {
+		allErrs = append(allErrs, errs.NewFieldRequired("masterURL"))
+	}
+
+	if len(config.MasterPublicURL) == 0 {
+		allErrs = append(allErrs, errs.NewFieldRequired("masterPublicURL"))
+	}
+
+	if len(config.AssetPublicURL) == 0 {
+		allErrs = append(allErrs, errs.NewFieldRequired("assetPublicURL"))
+	}
+
+	if config.SessionAuthenticationConfig != nil {
+		allErrs = append(allErrs, ValidateSessionAuthenticationConfig(config.SessionAuthenticationConfig).Prefix("sessionAuthenticationConfig")...)
+	}
+
+	allErrs = append(allErrs, ValidateGrantConfig(config.GrantConfig).Prefix("grantConfig")...)
+
+	for i, identityProvider := range config.IdentityProviders {
+		allErrs = append(allErrs, ValidateIdentityProvider(identityProvider).Prefix(fmt.Sprintf("identityProvider[%d]", i))...)
```

Comment:
if a provider is a password provider and has `UseAsLogin=true`, require `config.SessionAuthenticationConfig` to be non-null